### DISCUSSION TITLE
Status of XState for Mobile Platforms

### DISCUSSION THREAD

**Discussion by benkane** [statelyai/xstate#3315] [Q&A] [answered]
Hi!

I'm trying to tell if XState is still being ported to other platforms like Swift (and Kotlin if it was ever worked on). More specifically, I'm wondering about potential future support for native iOS and Android apps. Especially combined with the visualizations from https://stately.ai, it would be an amazing tool even outside the web world

I know Swift support was worked on at one point and was potentially going to be included in a V5 release along with other languages, but I haven't seen any news lately. Wondering if supporting other platforms is still a goal, if it's being worked on, if help is needed, etc.

Making a state machine is one thing. We can do that. But the combination of using XState's machine configs with https://stately.ai is where it's at. That combo can be really powerful, especially for teams.

Thanks!

**Comment by davidkpiano** [comment, marked as answer]
Hey, thanks for the inquiry!

Multi-language support is on the horizon, but our primary focus is XState v5 and the Stately Editor (currently in beta). XState v5 will be a good baseline for the features we want to support in other languages, and the general strategy is to port `@xstate/fsm` over to other languages first, which is a much easier task.

This will make it so that you can create state machines in various languages and have them be compatible with our tooling. Statechart support will be a stretch goal after that.
